python/graphstorm/eval/utils.py

```diff
@@ -233,7 +233,9 @@ def calc_ranking(pos_score, neg_score):
     _, indices = th.sort(scores, dim=1, descending=True)
     indices = th.nonzero(indices == 0)
     rankings = indices[:, 1].view(-1) + 1
-    rankings = rankings.cpu().detach()
+    rankings = rankings.detach()
+    if th.distributed.get_backend() == "gloo":
+        rankings = rankings.cpu() # Save GPU memory
```

Comment:
Does this mean for NCCL backed we want to store `rankings` on GPU? Is that necessary?

Reply:
We need to do all2all using nccl, so we have to save rankings on GPU. This may be a limitation of nccl.

Reply:
By default, its always stored on CPU right?

Reply:
The rankings are stored in GPU if the backend is nccl. If the backend is gloo, the rankings are stored in CPU.